pkg/addon/templateagent/decorator.go

```diff
@@ -329,6 +331,55 @@ func (d *imageDecorator) decorate(pod *corev1.PodTemplateSpec) error {
 	return nil
 }
 
+type proxyDecorator struct {
+	privateValues addonfactory.Values
+}
+
+func newProxyDecorator(privateValues addonfactory.Values) podTemplateSpecDecorator {
+	return &proxyDecorator{
+		privateValues: privateValues,
+	}
+}
+
+func (d *proxyDecorator) decorate(pod *corev1.PodTemplateSpec) error {
+	proxyConfig, ok := d.privateValues[ProxyPrivateValueKey]
+	if !ok {
+		return nil
+	}
+
+	pc, ok := proxyConfig.(addonapiv1alpha1.ProxyConfig)
+	if !ok {
+		return fmt.Errorf("proxy config value is invalid")
+	}
+
+	keyValues := []keyValuePair{}
+	if len(pc.HTTPProxy) > 0 {
+		keyValues = append(keyValues, keyValuePair{
+			name:  "HTTP_PROXY",
+			value: pc.HTTPProxy,
+		})
+	}
+	if len(pc.HTTPSProxy) > 0 {
+		keyValues = append(keyValues, keyValuePair{
+			name:  "HTTPS_PROXY",
+			value: pc.HTTPSProxy,
+		})
+	}
+	if len(pc.NoProxy) > 0 {
+		keyValues = append(keyValues, keyValuePair{
+			name:  "NO_PROXY",
+			value: pc.NoProxy,
+		})
+	}
```

Comment:
I'm not entirely sure whether the environment variables for proxy settings should be uppercase, lowercase, or if both are supported.

Reply:
It seems there is no standard and both uppercase and lowercase versions are used depending on the application

Reply:
both uppercase and lowercase are added.